pkg/controllers/awsloadbalancercontroller/webhooks.go

```diff
@@ -2,12 +2,371 @@ package awsloadbalancercontroller
+	if err != nil && errors.IsNotFound(err) {
+		return nil, false, nil
+	}
+	return &currentMWC, true, nil
+}
+
+func desiredValidatingWebhookConfiguration(controller *albo.AWSLoadBalancerController, webhookService *corev1.Service) *arv1.ValidatingWebhookConfiguration {
+	return &arv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-%s", controllerResourcePrefix, controller.Name),
+		},
+		Webhooks: []arv1.ValidatingWebhook{
+			{
+				Name: "vtargetgroupbinding.elbv2.k8s.aws",
+				ClientConfig: arv1.WebhookClientConfig{
+					Service: &arv1.ServiceReference{
+						Namespace: webhookService.Namespace,
+						Name:      webhookService.Name,
+						Path:      pointer.StringPtr("/validate-elbv2-k8s-aws-v1beta1-targetgroupbinding"),
+						Port:      pointer.Int32Ptr(controllerWebhookPort),
+					},
+				},
+				Rules: []arv1.RuleWithOperations{
+					{
+						Operations: []arv1.OperationType{
+							arv1.Create,
+							arv1.Update,
+						},
+						Rule: arv1.Rule{
+							APIGroups:   []string{"elbv2.k8s.aws"},
+							APIVersions: []string{"v1beta1"},
+							Resources:   []string{"targetgroupbindings"},
+						},
+					},
+				},
+				FailurePolicy:           failurePolicyPtr(arv1.Fail),
+				MatchPolicy:             matchPolicyPtr(arv1.Equivalent),
+				SideEffects:             sideEffectPtr(arv1.SideEffectClassNone),
+				AdmissionReviewVersions: []string{"v1beta1"},
+			},
+			{
+				Name: "vingress.elbv2.k8s.aws",
+				ClientConfig: arv1.WebhookClientConfig{
+					Service: &arv1.ServiceReference{
+						Namespace: webhookService.Namespace,
+						Name:      webhookService.Name,
+						Path:      pointer.StringPtr("/validate-networking-v1-ingress"),
+					},
+				},
+				Rules: []arv1.RuleWithOperations{
+					{
+						Operations: []arv1.OperationType{
+							arv1.Create,
+							arv1.Update,
+						},
+						Rule: arv1.Rule{
+							APIGroups:   []string{"networking.k8s.io"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"ingresses"},
+						},
+					},
+				},
+				FailurePolicy:           failurePolicyPtr(arv1.Fail),
+				MatchPolicy:             matchPolicyPtr(arv1.Equivalent),
+				SideEffects:             sideEffectPtr(arv1.SideEffectClassNone),
+				AdmissionReviewVersions: []string{"v1beta1"},
+			},
+		},
+	}
+}
+
+func sideEffectPtr(sideEffectClass arv1.SideEffectClass) *arv1.SideEffectClass {
+	return &sideEffectClass
+}
+
+func matchPolicyPtr(matchPolicyType arv1.MatchPolicyType) *arv1.MatchPolicyType {
+	return &matchPolicyType
+}
+
+func failurePolicyPtr(failurePolicyType arv1.FailurePolicyType) *arv1.FailurePolicyType {
+	return &failurePolicyType
+}
+
+func (r *AWSLoadBalancerControllerReconciler) updateValidatingWebhookConfiguration(ctx context.Context, current, desired *arv1.ValidatingWebhookConfiguration) error {
+	updatedVWC := current.DeepCopy()
+	var updated bool
+
+	if !haveValidatingWebhooksChanged(updatedVWC.Webhooks, desired.Webhooks) {
```

Comment:
Seems like the method name means the opposite of what it's supposed to.

Reply:
Refactored.